skada/_utils.py

```diff
@@ -59,11 +59,18 @@ def check_X_y_domain(
     X = check_array(X, input_name='X', allow_nd=allow_nd)
     y = check_array(y, force_all_finite=True, ensure_2d=False, input_name='y')
     check_consistent_length(X, y)
-    if sample_domain is None and allow_auto_sample_domain:
+    if sample_domain is None and not allow_auto_sample_domain:
+        raise ValueError("Either 'sample_domain' or 'allow_auto_sample_domain' "
+                         "should be set")
+    elif sample_domain is None and allow_auto_sample_domain:
+        y_type = _check_y_masking(y)
         sample_domain = np.ones_like(y)
         # labels masked with -1 are recognized as targets,
         # the rest is treated as a source
-        sample_domain[y == -1] = -2
+        if y_type == 'classification':
+            sample_domain[y == -1] = -2
```

Comment:
I always try to avoid using 'magic' numbers in the code, as they can be extremely difficult to understand and modify later on. Let's create a constant named `_DEFAULT_TARGET_DOMAIN_LABEL` = -2 (or something like this) in the current module namespace.